### ISSUE TITLE
Can't create asset

### ISSUE THREAD

**Issue by IpslWon** [snipe/snipe-it#1680]
When attempting to save a newly created asset. I get the following errors

ErrorException (E_NOTICE)
HELP
Undefined index: fields

Open: /var/www/snipeit/app/controllers/admin/AssetsController.php
        //attempt to validate
        $input=Input::all();
 
        $rules=$asset->validationRules();
        if($asset->model->fieldset)
        {
          foreach($asset->model->fieldset->fields AS $field) {
            $input[$field->db_column_name()]=$input['fields'][$field->db_column_name()];
            $asset->{$field->db_column_name()}=$input[$field->db_column_name()];
          }

Ubuntu 14.04
Snipe-IT version	v2.1.0-47-g39f40a3
PHP Version	5.5.9-1ubuntu4.14
Laravel Version	4.2.17

**Comment by snipe** [comment]
Can you give me more information about the asset you're trying to create? Does the model have custom fields associated with it?

**Comment by IpslWon** [comment]
It does have a custom filed associated with it. The field is the MAC address custom that came with the install.

**Comment by IpslWon** [comment]
Removing the custom field did the trick.

**Comment by uberbrady** [comment]
Still, that's weird - did you run all the migrations whenever you upgraded?

Did you try to 'detach' the MAC address field from the Mac Address Fieldset?

Do you know MySQL well enough to be able to do some simple queries for me? I'd love to track this down and figure out why it's happening; I don't want it to happen to anyone else.

**Comment by IpslWon** [comment]
This was a fresh install.

No, I don't think so. We're not using it much yet, so I can try to reproduce with a new model.

It's not completely foreign to me (MySQL). What do you have?

**Comment by IpslWon** [comment]
I created a test model with the MAC address custom field. I couldn't create an asset from it.

**Comment by uberbrady** [comment]
Well, I'd love to see the contents of:

`SELECT * FROM custom_fields`
and
`SELECT * FROM custom_fieldsets`
and maybe
`SELECT * FROM custom_field_custom_fieldset`

And while we're at it - 
`DESCRIBE assets`

Maybe from there we can see if there's some kind of data problem messing everything up.

**Comment by IpslWon** [comment]
```
mysql> SELECT * FROM custom_fields;
+----+-------------+-------------------------------------------------------------------------------------------+---------+---------------------+---------------------+---------+
| id | name        | format                                                                                    | element | created_at          | updated_at          | user_id |
+----+-------------+-------------------------------------------------------------------------------------------+---------+---------------------+---------------------+---------+
|  1 | MAC Address | [a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2}:[a-fA-F0-9]{2} | text    | 0000-00-00 00:00:00 | 0000-00-00 00:00:00 |    NULL |
+----+-------------+-------------------------------------------------------------------------------------------+---------+---------------------+---------------------+---------+
1 row in set (0.00 sec)
```

```
mysql> SELECT * FROM custom_fieldsets;
+----+------------------------+---------------------+---------------------+---------+
| id | name                   | created_at          | updated_at          | user_id |
+----+------------------------+---------------------+---------------------+---------+
|  1 | Asset with MAC Address | 0000-00-00 00:00:00 | 0000-00-00 00:00:00 |    NULL |
+----+------------------------+---------------------+---------------------+---------+
1 row in set (0.00 sec)
```

```
mysql> SELECT * FROM custom_field_custom_fieldset;
+-----------------+--------------------+-------+----------+
| custom_field_id | custom_fieldset_id | order | required |
+-----------------+--------------------+-------+----------+
|               1 |                  1 |     1 |        0 |
+-----------------+--------------------+-------+----------+
1 row in set (0.00 sec)
```

```
mysql> DESCRIBE assets;
+----------------------+---------------------------------------+------+-----+---------------------+----------------+
| Field                | Type                                  | Null | Key | Default             | Extra          |
+----------------------+---------------------------------------+------+-----+---------------------+----------------+
| id                   | int(10) unsigned                      | NO   | PRI | NULL                | auto_increment |
| name                 | varchar(255)                          | YES  |     | NULL                |                |
| asset_tag            | varchar(255)                          | NO   |     | NULL                |                |
| model_id             | int(11)                               | NO   |     | NULL                |                |
| serial               | varchar(255)                          | NO   |     | NULL                |                |
| purchase_date        | date                                  | YES  |     | NULL                |                |
| purchase_cost        | decimal(13,4)                         | NO   |     | 0.0000              |                |
| order_number         | varchar(255)                          | YES  |     | NULL                |                |
| assigned_to          | int(11)                               | YES  |     | NULL                |                |
| notes                | text                                  | YES  |     | NULL                |                |
| image                | text                                  | YES  |     | NULL                |                |
| user_id              | int(11)                               | NO   |     | NULL                |                |
| created_at           | timestamp                             | NO   |     | 0000-00-00 00:00:00 |                |
| updated_at           | timestamp                             | NO   |     | 0000-00-00 00:00:00 |                |
| physical             | tinyint(1)                            | NO   |     | 1                   |                |
| deleted_at           | timestamp                             | YES  |     | NULL                |                |
| status_id            | int(11)                               | YES  |     | NULL                |                |
| archived             | tinyint(1)                            | NO   |     | 0                   |                |
| warranty_months      | int(3)                                | YES  |     | NULL                |                |
| depreciate           | tinyint(1)                            | NO   |     | 0                   |                |
| supplier_id          | int(11)                               | YES  |     | NULL                |                |
| requestable          | tinyint(4)                            | NO   |     | 0                   |                |
| rtd_location_id      | int(11)                               | YES  |     | NULL                |                |
| _snipeit_mac_address | varchar(255)                          | YES  |     | NULL                |                |
| accepted             | enum('pending','accepted','rejected') | YES  |     | NULL                |                |
| last_checkout        | datetime                              | YES  |     | NULL                |                |
| expected_checkin     | date                                  | YES  |     | NULL                |                |
| company_id           | int(10) unsigned                      | YES  |     | NULL                |                |
+----------------------+---------------------------------------+------+-----+---------------------+----------------+
28 rows in set (0.00 sec)
```

**Comment by uberbrady** [comment]
Your database is completely pristine. Real mystery here.

Here's a thought - I wonder if you have stale views cached somewhere?

Was this an upgrade from a previous version, or a fresh install?

If it *was* an upgrade, try this: blow out the *contents* of the following directories (the directories themselves need to keep existing!) 

`app/storage/views/*`
`app/storage/cache/*`

**Comment by IpslWon** [comment]
This was a clean install.

**Comment by uberbrady** [comment]
Another thought - maybe the AJAX that loads in the custom fieldset somehow isn't loading? When you load up the Edit screen, an AJAX call fires to get the configuration/contents of the custom fields, so you can fill them out. I'm wondering if that AJAX call is failing, thus never rendering those fields - and making it so that no custom fields are submitted with the asset.

Or, maybe - if this is a particularly slow server - could it be that that AJAX call is taking a while and you're submitting the form beforehand?

Can you try and grab some of the information from: http://docs.snipeitapp.com/getting-help.html - so we can maybe figure out what's going on? The logs might be a good start.

**Comment by IpslWon** [comment]
I will try to get some more info, but this was a VM created only for this, so I don't think it's a resource issue. 

I also notice when I try to create an asset from a model with a custom field, the custom field doesn't show up anywhere.

**Comment by uberbrady** [comment]
Yeah, that definitely sounds like the issue. I bet if you can find a way to trace that internal AJAX call, you'll find it's 500'ing, and so you never get the chance to fill out your custom field.

I'd love to trace down why it's 500'ing - whatever you can find me in the logs could be really helpful. I assume it's a bug, and I'll try and fix it as quick as I can once we can figure out what it is.

**Comment by jmaliska** [comment]
Hi, we have the same problem in our environment. First I thought that this is migration issue, so now it is running on separate VM with clean installation of version 2.1.
Snipe-IT shows error 500 when creating asset with custom fields.

Log is attached here
[log-apache2handler-2016-01-26.txt](https://github.com/snipe/snipe-it/files/105694/log-apache2handler-2016-01-26.txt)


Snipe-IT version v2.1.0-47-g39f40a3
PHP Version 5.6.16
Laravel Version 4.2.17
OS is Oracle Linux 6.7
running on Apache httpd

**Comment by uberbrady** [comment]
Thanks for the log.

I suspect there's also a bug with the code that gets AJAX'ed in to populate the custom fields - *before* you even hit save.

Can you bring up the edit screen - don't hit save - and see if you have any other errors in the logs? Or, if you have any kind of web-dev tools, can you see if the server-side call is erroring?

I can't reproduce this locally, so it's hard for me to track down :(

**Comment by IpslWon** [comment]
I think I found the cause of my issue. 

The first part was I don't have the server name mapped to the IP. So the GET hardware request was erroring out

```
GET https://snipe-lm/hardware/models/5/custom_fields?_token=tHf6PzqE5vhxrBjje0nLWmIlWx4hqEFoj6vDmM8V net::ERR_NAME_NOT_RESOLVED
```

I then changes the app.php url config to the IP of the server, then I got a connection refused. I then chagned it to http (as oppose to https). Once I changed it to http when I selected the model with the custom field "MAC address" the MAC Address field showed up.
